OMCompiler/Compiler/runtime/System_omc.c

```diff
@@ -758,27 +758,35 @@ extern const char* System_snprintff(const char *fmt, int len, double d)
     MMC_THROW();
   }
   buf = ModelicaAllocateString(len);
-  if (snprintf(buf,len,fmt,d) >= len) {
+
+  int str_len = snprintf(buf, len, fmt, d);
+
+  if (str_len < 0) {
+    MMC_THROW();
+  }
+
+  if (str_len > len) {
     MMC_THROW();
   }
+
   return buf;
 }
 
 extern const char* System_sprintff(const char *fmt, double d)
 {
   char *buf;
-  const int buf_size = 20;
-  buf = ModelicaAllocateString(buf_size);
+  const int len = 20;
+  buf = ModelicaAllocateString(len);
 
-  int len = snprintf(buf, buf_size, fmt, d);
+  int str_len = snprintf(buf, len, fmt, d);
 
-  if (len < 0) {
+  if (str_len < 0) {
     MMC_THROW();
```

Comment:
If you are on it maybe add error messages to `System_sprintff` as well. Hunting throws without error messages can be a pain.